mps/04/cachelab.c

```diff
@@ -81,3 +81,57 @@ void registerTransFunction(void (*trans)(int M, int N, int[N][M], int[M][N]),
     func_list[func_counter].num_evictions =0;
     func_counter++;
 }
+
+struct TransposeInfo {
+    unsigned long long int A;
+    unsigned long long int B;
+    int M;
+    int N;
+};
+
+static int transposeInfoStatus = 0;
+static struct TransposeInfo transposeInfo;
+
+void initializeArrayAccessConverter(const char *tracefileName) {
+    char filename[4096];
+    sprintf(filename, "%s.info", tracefileName);
+    FILE *file = fopen(filename, "r");
+    if (file) {
+        int scanned = fscanf(file, "%llx %llx %d %d", &transposeInfo.A, &transposeInfo.B, &transposeInfo.M, &transposeInfo.N);
+        if (scanned == 4) {
```

Comment:
Should `fclose(file)` here (between lines 100 and 101)

Reply:
Good catch